server_src/lib/dataFunctions/printerClean.js

```diff
@@ -1548,48 +1548,21 @@ class PrinterClean {
   }
 
   static async initFarmInformation() {
-    farmStats = await FarmStatistics.find({});
-    if (typeof farmStats === undefined || farmStats.length < 1) {
+    farmStats = await FarmStatisticsService.list({});
+    if (typeof farmStats === "undefined" || farmStats.length < 1) {
       const farmStart = new Date();
-      const newfarmStats = new FarmStatistics({
-        farmStart,
-        heatMap,
-      });
-      farmStats[0] = newfarmStats;
-      newfarmStats.save();
+
+      farmStats[0] = await FarmStatisticsService.create(farmStart, heatMap);
     } else if (typeof farmStats[0].heatMap === "undefined") {
       farmStats[0].heatMap = heatMap;
       dashboardStatistics.utilisationGraph = heatMap;
       farmStats[0].markModified("heatMap");
-      farmStats[0].save();
-    } else {
-      heatMap = farmStats[0].heatMap;
```

Comment:
Deleted it @NotExpectedYet as it is duplicate, up to you to tell if this is creating problems down the line.
At least my test didnt detect a throw anymore, so the bug is gone.

Reply:
I think your right here. That extra save was probably an early attempt to fix another issue.

Reply:
Lol that makes me feel really uneasy

Reply:
I plan too, I wasn't meaning I wasn't going to haha